interceptor/interceptor.go

```diff
@@ -0,0 +1,221 @@
+type HandleQueryInput = internal.HandleQueryInput
+
+// WorkflowOutboundInterceptor is an interface for all workflow calls
+// originating from the SDK.
+//
+// All implementations must embed WorkflowOutboundInterceptorBase to safely
+// handle future changes.
+type WorkflowOutboundInterceptor = internal.WorkflowOutboundInterceptor
+
+// WorkflowOutboundInterceptorBase is a default implementation of
+// WorkflowOutboundInterceptor that forwards calls to the next outbound
+// interceptor.
+//
+// This must be embedded into all WorkflowOutboundInterceptor implementations to
+// safely handle future changes.
+type WorkflowOutboundInterceptorBase = internal.WorkflowOutboundInterceptorBase
+
+// ClientInterceptor for providing a ClientOutboundInterceptor to intercept
+// certain workflow-specific client calls from the SDK. If an implementation of
+// this is provided via client or worker options, certain client calls will be
+// intercepted by it.
+//
+// All implementations must embed ClientInterceptorBase to safely handle future
+// changes.
+type ClientInterceptor = internal.ClientInterceptor
+
+// ClientInterceptorBase is a default implementation of ClientInterceptor that
+// simply instantiates ClientOutboundInterceptorBase when called to intercept
+// the client.
+//
+// This must be embedded into all ClientInterceptor implementations to safely
+// handle future changes.
+type ClientInterceptorBase = internal.ClientInterceptorBase
+
+// ClientOutboundInterceptor is an interface for certain workflow-specific calls
+// originating from the SDK.
+//
+// All implementations must embed ClientOutboundInterceptorBase to safely handle
+// future changes.
+type ClientOutboundInterceptor = internal.ClientOutboundInterceptor
+
+// ClientOutboundInterceptorBase is a default implementation of
+// ClientOutboundInterceptor that forwards calls to the next outbound
+// interceptor.
+//
+// This must be embedded into all ActivityInboundInterceptor implementations to
+// safely handle future changes.
+type ClientOutboundInterceptorBase = internal.ClientOutboundInterceptorBase
+
+// ClientExecuteWorkflowInput is input for
+// ClientOutboundInterceptor.ExecuteWorkflow.
+type ClientExecuteWorkflowInput = internal.ClientExecuteWorkflowInput
+
+// ClientSignalWorkflowInput is input for
+// ClientOutboundInterceptor.SignalWorkflow.
+type ClientSignalWorkflowInput = internal.ClientSignalWorkflowInput
+
+// ClientSignalWithStartWorkflowInput is input for
+// ClientOutboundInterceptor.SignalWithStartWorkflow.
+type ClientSignalWithStartWorkflowInput = internal.ClientSignalWithStartWorkflowInput
+
+// ClientCancelWorkflowInput is input for
+// ClientOutboundInterceptor.CancelWorkflow.
+type ClientCancelWorkflowInput = internal.ClientCancelWorkflowInput
+
+// ClientTerminateWorkflowInput is input for
+// ClientOutboundInterceptor.TerminateWorkflow.
+type ClientTerminateWorkflowInput = internal.ClientTerminateWorkflowInput
+
+// ClientQueryWorkflowInput is input for
```

Comment:
All these "is input for" phrasings maybe read a bit better as "is the input to"

But you can take or leave it, not at all a big deal.

Reply:
Will change next commit

Reply:
Changed